src/lint/track_config.nim

```diff
@@ -283,6 +294,100 @@ proc checkExercisePrerequisites(trackConfig: TrackConfig;
                    "`slug` in the top-level `concepts` array"
         b.setFalseAndPrint(msg, path)
 
+proc statusMsg(exercise: ConceptExercise | PracticeExercise;
+               problem: string): string =
+  ## Returns the error text for an `exercise` with the status-related `problem`.
+  const exerciseKind =
+    when exercise is ConceptExercise:
+      "Concept Exercise"
+    else:
+      "Practice Exercise"
+  let statusStr =
+    case exercise.status
+    of sMissing:
+      "is user-facing (because a missing `status` key implies `active`)"
+    of sDeprecated:
+      "has a `status` of `deprecated`"
+    else:
+      &"is user-facing (because its `status` key has the value {q $exercise.status})"
+
+  result = &"The {exerciseKind} {q exercise.slug} {statusStr}" &
+           &", but has {problem}"
+
+proc checkExercisePCP(exercises: seq[ConceptExercise] | seq[PracticeExercise];
```

Comment:
What is `PCP`? To me this is a non-intuitive abbreviation so I'd prefer it being spelled out a bit more.

Reply:
> What is PCP?

Indeed. I'll try to improve it.

("practices, concepts, prerequisites")